pkg/cloudprovider/providers/azure/azure_backoff.go

```diff
@@ -102,14 +102,14 @@ func (az *Cloud) CreateOrUpdateLBWithRetry(lb network.LoadBalancer) error {
 }
 
 // CreateOrUpdatePIPWithRetry invokes az.PublicIPAddressesClient.CreateOrUpdate with exponential backoff retry
-func (az *Cloud) CreateOrUpdatePIPWithRetry(pip network.PublicIPAddress) error {
+func (az *Cloud) CreateOrUpdatePIPWithRetry(resourceGroup string, pip network.PublicIPAddress) error {
```

Comment:
Do we need to handle deletion if we create the PIP?  I believe we would delete it if we created it in the cluster RG, so probably we should delete if we create (but leave alone if the user created the PIP).  Would be good to get input from other reviewers on the preferred behaviour (I'm new to this too!).

Reply:
Good point. I added the code to handle deletion. Basically by matching both name and resource group.